### PR TITLE
[Test-1] Add test cases for ConfigFuture

### PR DESCRIPTION
**Pull request by mawer111**
It's my fitst PR,If some code wasn't good enough pls let me know.Thanks.😁

### CODE REVIEW

File: config/src/test/java/com/alibaba/fescar/config/ConfigFutureTest.java [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+    private final String content = "content";
+    private final String success = "success";
+
+    private ConfigFuture configFuture = new ConfigFuture(dataId, content, ConfigFuture.ConfigOperation.GET);
+
+    @Test
+    public void testGet() {
+        Assert.assertEquals(content, configFuture.get());
+        final SetResultTask setResultTask = new SetResultTask(configFuture, success);
+        new Thread(setResultTask).start();
```
Comment (jovany-wang) [inline config/src/test/java/com/alibaba/fescar/config/ConfigFutureTest.java]: Let's use lambda here?

File: config/src/test/java/com/alibaba/fescar/config/ConfigFutureTest.java [hunk 2]
```diff
@@ -17,26 +17,117 @@
...
+
+    @Test
+    public void testSetContent() {
+        configFuture.setContent(content);
+    }
+
+    @Test
+    public void testGetContent() {
+        configFuture.setContent(content);
+        Assert.assertEquals(content,configFuture.getContent());
```
Comment (jovany-wang) [inline config/src/test/java/com/alibaba/fescar/config/ConfigFutureTest.java]: `content, configFuture.getContent()`  
one space between 2 args.

File: config/src/test/java/com/alibaba/fescar/config/ConfigFutureTest.java [hunk 3]
```diff
@@ -0,0 +1,133 @@
...
+    }
+
+    @Test
+    public void testSetOperation() {
+        configFuture.setOperation(ConfigFuture.ConfigOperation.GET);
+    }
+
+    @Test
+    public void testIsTimeout() {
+        configFuture.isTimeout();
```
Comment (slievrly) [inline config/src/test/java/com/alibaba/fescar/config/ConfigFutureTest.java]: mock thread timeout？

File: config/src/test/java/com/alibaba/fescar/config/ConfigFutureTest.java [hunk 4]
```diff
@@ -0,0 +1,134 @@
...
+
+import java.util.concurrent.TimeUnit;
+
+/**
+ * @author mawerss1@gmail.com
+ * @Date 2019/03/7
+ */
+public class ConfigFutureTest {
+
+    private final String dataId = "dataId";
```
Comment (jovany-wang) [inline config/src/test/java/com/alibaba/fescar/config/ConfigFutureTest.java]: let's declare `dataId` as a `static final` variable.

File: config/src/test/java/com/alibaba/fescar/config/ConfigFutureTest.java [hunk 5]
```diff
@@ -0,0 +1,134 @@
...
+
+import java.util.concurrent.TimeUnit;
+
+/**
+ * @author mawerss1@gmail.com
+ * @Date 2019/03/7
+ */
+public class ConfigFutureTest {
+
+    private static final String dataId = "dataId";
```
Comment (slievrly) [inline config/src/test/java/com/alibaba/fescar/config/ConfigFutureTest.java]: static final variables should be toUpper + _

File: config/src/test/java/com/alibaba/fescar/config/ConfigFutureTest.java [hunk 6]
```diff
@@ -0,0 +1,134 @@
...
+     * Test get with timeout argument.
+     */
+    @Test
+    public void testGetWithTimeOut() {
+        long outTime = 200L;
+        new Thread(() -> {
+            try {
+                Thread.sleep(outTime + 100);
+            } catch (InterruptedException e) {
+                e.printStackTrace();
```
Comment (xingfudeshi) [inline config/src/test/java/com/alibaba/fescar/config/ConfigFutureTest.java]: Do not print stack trace.

Comment (mawer111) [reply config/src/test/java/com/alibaba/fescar/config/ConfigFutureTest.java]: done

File: config/src/test/java/com/alibaba/fescar/config/ConfigFutureTest.java [hunk 7]
```diff
@@ -0,0 +1,134 @@
...
+        Assert.assertEquals(Boolean.FALSE, configFuture.get());
+        startSetResultThread(RESULT);
+        Assert.assertEquals(RESULT, configFuture.get());
+    }
+
+    /**
+     * Test get with timeout argument.
+     */
+    @Test
+    public void testGetWithTimeOut() {
```
Comment (xingfudeshi) [inline config/src/test/java/com/alibaba/fescar/config/ConfigFutureTest.java]: throws Exception

File: config/src/test/java/com/alibaba/fescar/config/ConfigFutureTest.java [hunk 8]
```diff
@@ -0,0 +1,134 @@
...
+        long outTime = 200L;
+        new Thread(() -> {
+            try {
+                Thread.sleep(outTime + 100);
+            } catch (InterruptedException e) {
+
```
Comment (xingfudeshi) [inline config/src/test/java/com/alibaba/fescar/config/ConfigFutureTest.java]: throw the exception out

File: config/src/test/java/com/alibaba/fescar/config/ConfigFutureTest.java [hunk 9]
```diff
@@ -0,0 +1,134 @@
...
+            }
+            configFuture.setResult(RESULT);
+        });
```
Comment (xingfudeshi) [inline config/src/test/java/com/alibaba/fescar/config/ConfigFutureTest.java]: forget to start the thread?

File: config/src/test/java/com/alibaba/fescar/config/ConfigFutureTest.java [hunk 10]
```diff
@@ -0,0 +1,134 @@
...
+    }
+
+    @Test
+    public void testGetOperation() {
+        configFuture.setOperation(ConfigFuture.ConfigOperation.GET);
+        Assert.assertEquals(ConfigFuture.ConfigOperation.GET, configFuture.getOperation());
+    }
+
+    @Test
+    public void testSetOperation() {
```
Comment (slievrly) [inline config/src/test/java/com/alibaba/fescar/config/ConfigFutureTest.java]: assert something？

File: config/src/test/java/com/alibaba/fescar/config/ConfigFutureTest.java [hunk 11]
```diff
@@ -0,0 +1,134 @@
...
+
+            }
+            configFuture.setResult(RESULT);
+        });
+        Assert.assertEquals(CONTENT, configFuture.get(outTime, TimeUnit.MILLISECONDS));
+    }
+
+    @Test
+    public void testSetContent() {
+        configFuture.setContent(CONTENT);
```
Comment (slievrly) [inline config/src/test/java/com/alibaba/fescar/config/ConfigFutureTest.java]: assert something

File: config/src/test/java/com/alibaba/fescar/config/ConfigFutureTest.java [hunk 12]
```diff
@@ -0,0 +1,139 @@
...
+     * Test get with timeout argument.
+     */
+    @Test
+    public void testGetWithTimeOut() {
+        long outTime = 50L;
+        new Thread(() -> {
+            final long start = System.currentTimeMillis();
+            while (true) {
+                //mock timeout
+                if ((System.currentTimeMillis() - start) > outTime * 1000) {
```
Comment (slievrly) [inline config/src/test/java/com/alibaba/fescar/config/ConfigFutureTest.java]: 50s？

**Comment by slievrly** [comment]
@jovany-wang

File: config/src/test/java/com/alibaba/fescar/config/ConfigFutureTest.java [hunk 13]
```diff
@@ -0,0 +1,139 @@
...
+        Assert.assertEquals(RESULT, configFuture.get());
+    }
+
+    /**
+     * Test get with put operation.
+     */
+    @Test
+    public void testGetWithPutOperation() {
+        configFuture.setOperation(ConfigFuture.ConfigOperation.PUT);
+        Assert.assertEquals(Boolean.FALSE, configFuture.get());
```
Comment (jovany-wang) [inline config/src/test/java/com/alibaba/fescar/config/ConfigFutureTest.java]: assertFalse

File: config/src/test/java/com/alibaba/fescar/config/ConfigFutureTest.java [hunk 14]
```diff
@@ -0,0 +1,139 @@
...
+        startSetResultThread(RESULT);
+        Assert.assertEquals(RESULT, configFuture.get());
+    }
+
+    /**
+     * Test get with timeout argument.
+     */
+    @Test
+    public void testGetWhenTimeOut() {
+        long outTime = 50L;
```
Comment (jovany-wang) [inline config/src/test/java/com/alibaba/fescar/config/ConfigFutureTest.java]: `final long outTime = 50L;`
or declare it as a private static final variable.

File: config/src/test/java/com/alibaba/fescar/config/ConfigFutureTest.java [hunk 15]
```diff
@@ -0,0 +1,139 @@
...
+            final long start = System.currentTimeMillis();
+            while (true) {
+                //mock timeout
+                if ((System.currentTimeMillis() - start) > outTime * 2) {
+                    break;
+                }
+            }
+            configFuture.setResult(RESULT);
+        }).start();
+        Assert.assertEquals(CONTENT, configFuture.get(outTime, TimeUnit.MILLISECONDS));
```
Comment (jovany-wang) [inline config/src/test/java/com/alibaba/fescar/config/ConfigFutureTest.java]: should we `join()` the sub-thread here?

File: config/src/test/java/com/alibaba/fescar/config/ConfigFutureTest.java [hunk 16]
```diff
@@ -0,0 +1,139 @@
...
+    @Test
+    public void testSetOperation() {
+        configFuture.setOperation(ConfigFuture.ConfigOperation.GET);
+        Assert.assertEquals(ConfigFuture.ConfigOperation.GET, configFuture.getOperation());
+    }
+
+    @Test
+    public void testIsTimeout() throws InterruptedException {
+        configFuture.get(TIME_MILLS, TimeUnit.MILLISECONDS);
+        Thread.sleep(TIME_MILLS * 2);
```
Comment (jovany-wang) [inline config/src/test/java/com/alibaba/fescar/config/ConfigFutureTest.java]: TimeUtil.sleep?

File: config/src/test/java/com/alibaba/fescar/config/ConfigFutureTest.java [hunk 17]
```diff
@@ -0,0 +1,139 @@
...
+        Assert.assertTrue(configFuture.isTimeout());
+    }
+
+    private void startSetResultThread(Object result) {
+        new Thread(() -> configFuture.setResult(result)).start();
```
Comment (jovany-wang) [inline config/src/test/java/com/alibaba/fescar/config/ConfigFutureTest.java]: also join the sub-thread

File: config/src/test/java/com/alibaba/fescar/config/ConfigFutureTest.java (hunk 13 above)
Comment (mawer111) [reply config/src/test/java/com/alibaba/fescar/config/ConfigFutureTest.java]: configFuture.get() return Object, use assertFalse need type cast

**Comment by xingfudeshi** [comment]
@jovany-wang

**Pull request merged**